openmdao/components/exec_comp.py

```diff
@@ -193,18 +223,21 @@ def __init__(self, exprs, vectorize=False, units=None, **kwargs):
         self._exprs = exprs[:]
         self._codes = None
         self._kwargs = kwargs
-        self._vectorize = vectorize
-        self._units = units
 
     def setup(self):
         """
         Set up variable name and metadata lists.
         """
+        if not self._exprs:
```

Comment:
Can this ever happen? I guess if they pass in `None` as their expression list.  Raising this in `__init__` instead of `setup` might be more helpful to the user.

Reply:
This was a change I made (see the new test) to enable calling the constructor with no args, which is necessary for `embed-options` to display the options table.